pkg/controller/garbagecollector/memcachediscovery/client.go

```diff
@@ -0,0 +1,167 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package memcachediscovery includes a Client which is a CachedDiscoveryInterface.
+package memcachediscovery
```

Comment:
How about getting this merged separately and see about making use of it in the namespace cleanup controller?  It's a smaller controller and it would reduce the size of this pull.